docs/jsx.md

```diff
@@ -183,6 +184,34 @@ You can use hooks in your production environment to run the production build scr
 }
 ```
 
+#### Making `m` accessible globally
+In order to access `m` globally from all your project first import `webpack` in your `webpack.config.js` like this:
+```
+const webpack = require('webpack')
+```
+Then create a new plugin in the `plugins` property of the Webpack configuration object:
+```js
+{
+	plugins: [
+		new webpack.ProvidePlugin({
+			m: 'mithril'
+		})
+	]
+}
+```
+See [the Webpack docs](https://webpack.js.org/plugins/provide-plugin/) for more information on `ProvidePlugin`.
```

Comment:
Could you strike this entire section and just append to both the ["Setup"](https://mithril.js.org/jsx.html#setup) and the ["Using Babel with Webpack"](https://mithril.js.org/jsx.html#using-babel-with-webpack) sections simple starter `/src/index.js` + `/index.html` files for each, so it's clear from example what the code would look like in practice? Specifically:

- In "Setup", append something like this after the `.babelrc`:

	> Next, create a file called `src/index.js`:
	>
	> ```jsx
	> ;(function () {
	> "use strict"
	> 
	> function App() {
	> 	return {
	> 		view: () => (
	> 			<div>Hello, world</div>
	> 		)
	> 	}
	> }
	>
	> m.mount(document.body, App)
	> })()
	> ```
	> 
	> Next, create a file called `index.html`:
	> 
	> ```html
	> <!DOCTYPE html>
	> <meta charset="utf-8">
	> <title>Hello, world!</title>
	> <script src="path/to/mithril.js"></script>
	> <script src="bin/index.js"></script>
	> ```

- In "Using Babel with Webpack", append something like this after the `webpack.config.js`

	> Next, create a file called `src/index.js`:
	>
	> ```jsx
	> import m from "mithril"
	> 
	> function App() {
	> 	return {
	> 		view: () => (
	> 			<div>Hello, world</div>
	> 		)
	> 	}
	> }
	>
	> m.mount(document.body, App)
	> ```
	> 
	> Next, create a file called `index.html`:
	> 
	> ```html
	> <!DOCTYPE html>
	> <meta charset="utf-8">
	> <title>Hello, world!</title>
	> <script src="bin/app.js"></script>
	> ```

	Note: we want to *encourage* local use when using bundlers - while we support global use and of course you can use the provide plugin for Webpack or rollup-plugin-inject for Rollup, we show a clear idiomatic preference for imported use throughout the documentation when you're using a bundler, and it makes tracing Mithril code a lot easier.

Reply:
> we want to encourage local use when using bundlers

Shouldn't Mithril's docs be agnostic about how to use third party libraries?

Reply:
We're agnostic about others, but it's a recommendation for a reason. It's way too easy to lose track of what uses what when you use globals, especially if you aren't using TypeScript.

Also, even spec writers favor using ES modules + bundlers over globals, and some proposals specifically require `import` to access (like [WICG's `<virtual-scroller>`](https://github.com/WICG/virtual-scroller) and [TC39's `temporal`](https://github.com/tc39/proposal-temporal)), and [TC39 is planning to reserve `js:` for this purpose](https://github.com/tc39/proposal-javascript-standard-library/). So such an opinion isn't exactly far-fetched.

Reply:
I agree on principle, but I don't think it's such a bad idea to include the global config as a footnote. It seems a bit far fetched to think that anyone would lose track of Mithril when it is used in most files in a project.

Also Mithril itself favors a global `m` object much like jQuery's global `$`. It seems a bit contradictory to at least not inform the user on how to achieve the same thing using Webpack.

Anyway, it's not my place to against the grain here. I'll let you know when I have made the changes you've suggested.

Reply:
I'll note our non-bundling setup already features Mithril globally loaded via script. So it's not like we ignored it altogether. Also, both Webpack (natively) and Rollup (via plugin) support injecting globals from modules - it's how they support Node.js globals when compiling to browsers.

Reply:
> I'll note our non-bundling setup already features Mithril globally loaded via script

That's precisely my point :)

I'm back from holidays. I'll let you know when I've implemented the changes you requested.